ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -672,28 +689,26 @@ runThreadNetwork ThreadNetworkArgs
 
       blockProduction <- blockProductionIOLike pInfoConfig canBeLeader maintainForgeState varRNG $
          \upd currentBno tickedLdgSt txs prf -> do
-            let currentSlot = tickedSlotNo tickedLdgSt
-
-            -- the typical behavior, which doesn't add a Just-In-Time EBB
-            let forgeWithoutEBB =
+            let currentSlot  = tickedSlotNo tickedLdgSt
+            let currentEpoch = HFF.futureSlotToEpoch future currentSlot
+
+            -- EBBs are only ever possible in the first era
+            let inFirstEra = HFF.futureEpochInFirstEra future currentEpoch
+
+            let ebbSlot :: SlotNo
+                ebbSlot = SlotNo $ x * y
+                  where
+                    EpochNo   x = currentEpoch
+                    EpochSize y = epochSize0
+            let p = ledgerTipPoint $ tickedLedgerState tickedLdgSt
+
+            let needEBB = inFirstEra && At ebbSlot > pointSlot p
+            case mbForgeEbbEnv <* guard needEBB of
+              Nothing ->
+                 -- no EBB needed, forge without making one
                   forgeBlock pInfoConfig upd
                     currentBno tickedLdgSt txs prf
-
```

Comment:
This logic was moved to the add-chunk above. Slightly reorganized, but the only difference is the new `inFirstEra` check.